pkg/resource/account/sdk.go

```diff
@@ -0,0 +1,377 @@
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package account
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackcondition "github.com/aws-controllers-k8s/runtime/pkg/condition"
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+	ackrequeue "github.com/aws-controllers-k8s/runtime/pkg/requeue"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	"github.com/aws/aws-sdk-go/aws"
+	svcsdk "github.com/aws/aws-sdk-go/service/organizations"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	svcapitypes "github.com/aws-controllers-k8s/organizations-controller/apis/v1alpha1"
+)
+
+// Hack to avoid import errors during build...
+var (
+	_ = &metav1.Time{}
+	_ = strings.ToLower("")
+	_ = &aws.JSONValue{}
+	_ = &svcsdk.Organizations{}
+	_ = &svcapitypes.Account{}
+	_ = ackv1alpha1.AWSAccountID("")
+	_ = &ackerr.NotFound
+	_ = &ackcondition.NotManagedMessage
+	_ = &reflect.Value{}
+	_ = fmt.Sprintf("")
+	_ = &ackrequeue.NoRequeue{}
+)
+
+// sdkFind returns SDK-specific information about a supplied resource
+func (rm *resourceManager) sdkFind(
+	ctx context.Context,
+	r *resource,
+) (latest *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.sdkFind")
+	defer func() {
+		exit(err)
+	}()
+	// If any required fields in the input shape are missing, AWS resource is
+	// not created yet. Return NotFound here to indicate to callers that the
+	// resource isn't yet created.
+	if rm.requiredFieldsMissingFromReadOneInput(r) {
+		return nil, ackerr.NotFound
+	}
+
+	input, err := rm.newDescribeRequestPayload(r)
+	if err != nil {
+		return nil, err
+	}
+
+	var resp *svcsdk.DescribeAccountOutput
+	resp, err = rm.sdkapi.DescribeAccountWithContext(ctx, input)
+	rm.metrics.RecordAPICall("READ_ONE", "DescribeAccount", err)
+	if err != nil {
+		if reqErr, ok := ackerr.AWSRequestFailure(err); ok && reqErr.StatusCode() == 404 {
+			return nil, ackerr.NotFound
+		}
+		if awsErr, ok := ackerr.AWSError(err); ok && awsErr.Code() == "UNKNOWN" {
+			return nil, ackerr.NotFound
+		}
+		return nil, err
+	}
+
+	// Merge in the information we read from the API call above to the copy of
+	// the original Kubernetes object we passed to the function
+	ko := r.ko.DeepCopy()
+
+	if ko.Status.ACKResourceMetadata == nil {
+		ko.Status.ACKResourceMetadata = &ackv1alpha1.ResourceMetadata{}
+	}
+	if resp.Account.Arn != nil {
+		arn := ackv1alpha1.AWSResourceName(*resp.Account.Arn)
+		ko.Status.ACKResourceMetadata.ARN = &arn
+	}
+	if resp.Account.Email != nil {
+		ko.Spec.Email = resp.Account.Email
+	} else {
+		ko.Spec.Email = nil
+	}
+	if resp.Account.Id != nil {
+		ko.Status.ID = resp.Account.Id
+	} else {
+		ko.Status.ID = nil
+	}
+
```

Comment:
We need to requeue in here if state is `IN_PROGRESS`. Here is an example on how it's done for EKS Clusters https://github.com/aws-controllers-k8s/eks-controller/blob/main/templates/hooks/cluster/sdk_read_one_post_set_output.go.tpl#L1-L7